server/planning/assignments/assignments.py

```diff
@@ -355,6 +355,10 @@ def send_assignment_notification(self, updates, original=None, force=False):
         if assigned_to_user and get_user_notification_preferences(assigned_to_user, "assignments")["email"] is False:
             return
 
+        # No assignment desktop notification sent, if a user has disabled their global desktop notifications
+        if not assigned_to_user.get("user_preferences", {}).get("desktop:notification", {}).get("enabled", True):
```

Comment:
If the user has an assignment mail toggle is on and desktop notification is off, so in this case nothing is going to happen, means user has not received any mail.